### PR TITLE
14789 Create UI throwing schema error when court order is not entered

### PR DESCRIPTION
**Pull request by jonathan-longe**
*Issue #:* /bcgov/entity#14789

*Description of changes:*

Fix bug that caused schema error when submitting an incorporation application without a court order.

14789_court_order_error_D

Here's the relevant snippet of the payload that was causing the error:
```    
"courtOrder": {
      "fileNumber": "",
      "hasPlanOfArrangement": false
}
```

The fix removes the "courtOrder" attribute unless either the fileNumber is populated or "hasPlanOfArrangement" is true.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/mixins/filing-template-mixin.ts [hunk 1]
```diff
@@ -200,11 +200,14 @@ export default class FilingTemplateMixin extends DateMixin {
         }
 
         if (this.isTypeBcUlcCompany) {
+          // Add Court Order ONLY when it is required and applied.
           const courtOrder = this.getCourtOrderStep.courtOrder
-          filing.incorporationApplication.courtOrder = {
-            fileNumber: courtOrder.fileNumber,
-            effectOfOrder: courtOrder.hasPlanOfArrangement ? EffectOfOrders.PLAN_OF_ARRANGEMENT : '',
-            hasPlanOfArrangement: courtOrder.hasPlanOfArrangement
+          if (courtOrder.hasPlanOfArrangement || courtOrder.fileNumber) {
```
Comment (jonathan-longe) [inline src/mixins/filing-template-mixin.ts:205]: The courtOrder attribute is only populated if the court order number is provided or hasPlanOfArrangement is checked.

File: src/store/getters/state-getters.ts [hunk 2]
```diff
@@ -491,7 +491,8 @@ export const isIncorporationApplicationValid = (state: StateIF): boolean => {
   const isBaseStepsValid = (
     getCreateShareStructureStep(state).valid &&
     getEffectiveDateTime(state).valid &&
-    getIncorporationAgreementStep(state).valid
+    getIncorporationAgreementStep(state).valid &&
+    getCourtOrderStep(state).valid
```
Comment (jonathan-longe) [inline src/store/getters/state-getters.ts:495]: The getCourtOrderStep() method was not being called to check for validation.

File: src/views/Incorporation/IncorporationReviewConfirm.vue [hunk 3]
```diff
@@ -142,7 +142,7 @@
           :autoValidation="getValidateSteps"
           :draftCourtOrderNumber="getCourtOrderStep.courtOrder.fileNumber"
           :hasDraftPlanOfArrangement="getCourtOrderStep.courtOrder.hasPlanOfArrangement"
-          :courtOrderNumberRequired="true"
```
Comment (jonathan-longe) [inline src/views/Incorporation/IncorporationReviewConfirm.vue:145]: The court order number is not required unless hasPlanOfArrangment is checked

File: src/store/getters/state-getters.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/store/getters/state-getters.ts:495]: You found it. Nice!

Now this is checked for all "base" entity types (BEN, BC, CC, ULC). Is this step valid for those other types? You might need a conditional here.

**Review by severinbeauvais — COMMENTED** [review]
Looks good. Please see comments and update the app version (and check in those files):
- package.json
- package-lock.json

**Comment by severinbeauvais** [comment]
Ohhhh, I see some failing unit tests.

You can run them locally using `npm run test:unit`. (I usually redirect the output to a file so I can look at it at my leisure.)

**Comment by jonathan-longe** [comment]
@severinbeauvais thanks for your comments.  I've incremented the version number and fixed the unit test errors

File: tests/unit/Actions.spec.ts [hunk 4]
```diff
@@ -143,6 +143,7 @@ describe('Emits error event if NR validation fails in file and pay', () => {
     store.state.stateModel.createShareStructureStep = { valid: true }
     store.state.stateModel.incorporationAgreementStep = { valid: true }
     store.state.stateModel.effectiveDateTime = { valid: true }
+    store.state.stateModel.courtOrderStep = { valid: true }
```
Comment (severinbeauvais) [inline tests/unit/Actions.spec.ts:146]: The tests are working exactly as intended 👍

File: tests/unit/IncorporationReviewConfirm.spec.ts [hunk 5]
```diff
@@ -67,8 +67,7 @@ for (const test of reviewConfirmTestCases) {
         IncorporationResources
       )
 
-      const expected = (test.entityType === 'ULC')
-      expect(wrapper.find('#court-order-poa-section').exists()).toBe(test.entityType === 'ULC')
+      expect(wrapper.find('#court-order-poa-section').exists()).toBe(['BEN', 'ULC', 'CC', 'BC'].includes(test.entityType))
```
Comment (severinbeauvais) [inline tests/unit/IncorporationReviewConfirm.spec.ts:70]: Oooo, fancy, haha. 😃 👍

File: src/store/getters/state-getters.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/store/getters/state-getters.ts:497]: The court order component is now rendered for BEN/BC/CCC/ULC, so this check should be valid for all "base" types.
= resolved

**Review by severinbeauvais — APPROVED** [review]

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by ketaki-deodhar — APPROVED** [review]

**Comment by severinbeauvais** [comment]
Do not merge until testing-coverage check has passed.

File: src/components/common/Actions.vue [hunk 6]
```diff
@@ -174,6 +174,7 @@ export default class Actions extends Vue {
       // clear flag
       this.setHaveChanges(false)
     } catch (error) {
+      console.log('Error on onClickSave(): ', error)
```
Comment (severinbeauvais) [inline src/components/common/Actions.vue:177]: Is this the only catch block in here where the error isn't logged?

File: src/mixins/filing-template-mixin.ts [hunk 7]
```diff
@@ -82,7 +82,7 @@ export default class FilingTemplateMixin extends DateMixin {
   @Getter getStaffPaymentStep!: StaffPaymentStepIF
   @Getter getCourtOrderStep!: CourtOrderStepIF
   @Getter isRoleStaff!: boolean
-  @Getter isTypeBcUlcCompany!: boolean
+  @Getter isBaseCompany!: boolean
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:85]: No longer needed?

**Comment by severinbeauvais** [comment]
If you have no further code changes and the tests pass, I am OK with you merging this.

File: src/mixins/filing-template-mixin.ts (hunk 7 above)
Comment (jonathan-longe) [reply src/mixins/filing-template-mixin.ts:85]: That's true, good catch!  I'll fix it.

File: src/components/common/Actions.vue (hunk 6 above)
Comment (jonathan-longe) [reply src/components/common/Actions.vue:177]: Good thinking, I've added console.log statements to the catch blocks.  Will push a new commit shortly.

Comment (severinbeauvais) [reply src/components/common/Actions.vue:177]: Thanks.

**Pull request merged**